client/client-multi/client-objective-c/KaaTests/event/registration/DefaultEndpointRegistrationManagerTest.m

```diff
@@ -242,6 +247,29 @@ - (void)testCheckOnDetachedDelegate {
     [verifyCount(state, times(2)) setIsAttachedToUser:NO];
 }
 
+- (void)testCheckAddDetachDelegate {
+    id<KaaClientState> state = mockProtocol(@protocol(KaaClientState));
+    [given([state endpointAccessToken]) willReturn:@""];
+
+    EndpointDetachResponse *detachResponse1 = [[EndpointDetachResponse alloc] init];
+    detachResponse1.requestId = 0;
+    detachResponse1.result = SYNC_RESPONSE_RESULT_TYPE_SUCCESS;
+    
+    id<EndpointAttachDetachDelegate> delegate = mockProtocol(@protocol(EndpointAttachDetachDelegate));
+    
+    DefaultEndpointRegistrationManager *manager = [[DefaultEndpointRegistrationManager alloc] initWithState:state executorContext:self.executorContext userTransport:nil profileTransport:nil];
+    
+    EndpointKeyHash *emptyKeyHash1 = [[EndpointKeyHash alloc] initWithKeyHash:nil];
+
+    [manager detachEndpointWithKeyHash:emptyKeyHash1 delegate:delegate];
+    
+    [manager onUpdateWithAttachResponses:nil detachResponses:@[detachResponse1] userResponse:nil userAttachNotification:nil userDetachNotification:nil];
+    
+    [NSThread sleepForTimeInterval:1.f];
+    
+    [verifyCount(delegate, times(1)) onDetachResult:SYNC_RESPONSE_RESULT_TYPE_SUCCESS];
```

Comment:
There should exist a method verifyCount with timeout so that "sleepForTimeInterval" wouldn't be needed.